plugins/modules/api_modify.py

```diff
@@ -539,6 +540,22 @@ def sync_list(module, api, path, path_info):
                         index=index + 1,
                     )
                 )
+        for require_list in require_one_of:
```

Comment:
I think this should happen in `polish_entry()`, so that `require_one_of` is available for all synchronization types.